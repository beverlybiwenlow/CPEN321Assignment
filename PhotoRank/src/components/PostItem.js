import React, { Component } from 'react';
import { Text, View, Image } from 'react-native';

import { Card, CardSection } from './common';

class PostItem extends Component {
    render() {
        const { thumbnailStyle, headerContentStyle, footerContentStyle, thumbnailContainerStyle, headerTextStyle, imageStyle } = styles;
        const { post } = this.props
        return (
            <Card>
                <CardSection>
                  <View style = {thumbnailContainerStyle}>
                    <Image style={thumbnailStyle} source={{uri: "https://i.pinimg.com/736x/28/64/d5/2864d5114f4f7be2abef0fceb6ccb7c2--funny-mugshots-mug-shots.jpg"}} />
                  </View>
                  <View style = {headerContentStyle}>
                    <Text style = {headerTextStyle}>
                        { "User: " + post.displayName }
                    </Text>
                    <Text>Company name</Text>
                  </View>
                </CardSection>
                <CardSection>
                  <Image style={imageStyle} source={{ uri: "https://iso.500px.com/wp-content/uploads/2014/04/20482.jpg" }} />
                </CardSection>
                <CardSection>
                  <View style = {footerContentStyle}>
                    <Text>
                        { 'Likes: ' + post.likeCount }
                    </Text>
                    <Text>
                        { 'Caption: ' + post.caption }
                    </Text>
                    <Text>
                        {'Url: ' + post.url}
                    </Text>
                  </View>
                </CardSection>
            </Card>
        );
    }
}

const styles = {
  headerContentStyle: {
    flexDirection: 'column',
    justifyContent: 'space-around'
  },
  footerContentStyle: {
    flexDirection: 'column',
    justifyContent: 'space-around'
  },
  headerTextStyle: {
    fontSize: 18
  },
  thumbnailStyle: {
    height: 50,
    width: 50
  },
  thumbnailContainerStyle: {
    justifyContent: 'center',
    alignItems: 'center',
    marginLeft: 10,
    marginRight: 10
  },
  imageStyle: {
    height: 300,
    flex: 1,
    width: null
  }
};

export default PostItem;
