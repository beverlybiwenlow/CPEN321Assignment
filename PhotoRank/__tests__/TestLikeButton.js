// Link.react-test.js
import React from 'react';
import PostItem from '../src/components/PostItem';
import ErrorBoundary from '../src/components/ErrorBoundary';
import renderer from 'react-test-renderer';


// const React = require('React');
// const ReactTestRenderer = require('react-test-renderer');
// const Text = require('Text');
// const TouchableHighlight = require('TouchableHighlight');

// describe('TouchableHighlight', () => {
//   it('renders correctly', () => {
//     const instance = ReactTestRenderer.create(
//         <PostItem/>
//         // <Text>HI</Text>
//     );
//     console.log(instance.props);
//     expect(instance.toJSON()).toMatchSnapshot();
//   });
// });

test('Like button toggles the class when clicked', () => {
  const component = renderer.create(
    // <ErrorBoundary>
        <PostItem/>
    // </ErrorBoundary>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.onButtonPress();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.onButtonPress();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});