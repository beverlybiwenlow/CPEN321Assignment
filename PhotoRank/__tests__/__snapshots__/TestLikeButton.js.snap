// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Like button toggles the class when clicked 1`] = `
<h1>
  Something went wrong.
</h1>
`;

exports[`TouchableHighlight renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
>
  HI
</Text>
`;
